src/pages/Playlists/playlists-reducer.js

```diff
@@ -0,0 +1,73 @@
+export const initialPlaylists = [];
+export const SET_PLAYLISTS = "setPlaylists";
+export const ADD_VIDEO_TO_PLAYLIST = "addVideoToPlaylist";
+export const REMOVE_VIDEO_FROM_PLAYLIST = "removeVideofromPlaylist";
+export const CREATE_PLAYLIST = "createPlaylist";
+export const DELETE_PLAYLIST = "deletePlaylist";
+export const EDIT_PLAYLIST_NAME = "editPlaylistName";
+export const SET_LIKED_VIDEOS_ID = "setLikedVideosId";
+export const SET_WATCH_LATER_ID = "setWatchLaterId";
+
+export const playlistsReducer = (
+  state,
+  { type, playlist, playlistId, playlistName, videoId, video, fetchedPlaylists }
+) => {
+  const { playlists } = state;
+  switch (type) {
+    case SET_PLAYLISTS:
+      return { ...state, playlists: fetchedPlaylists };
+    case CREATE_PLAYLIST:
+      return {
+        ...state,
+        playlists: playlists.concat(playlist),
+      };
+    case DELETE_PLAYLIST:
+      return {
+        ...state,
+        playlists: playlists.filter((playlist) => playlist._id !== playlistId),
+      };
+    case ADD_VIDEO_TO_PLAYLIST:
+      return {
+        ...state,
+        playlists: playlists.map((playlist) =>
+          playlist._id !== playlistId
+            ? playlist
+            : { ...playlist, videos: playlist.videos.concat(video) }
+        ),
+      };
```

Comment:
Extract this and similar reducer actions to another function and use them, 
And you can refactor it to this in ADD_VIDEO_ && REMOVE_VIDEO_
```suggestion
        playlists: playlists.map((playlist) =>
          playlist._id === playlistId
            ?  { ...playlist, videos: playlist.videos.concat(video) } : playlist            
        ),
      };
```